clang/include/clang/AST/DeclBase.h

```diff
@@ -817,11 +817,11 @@ class alignas(8) Decl {
            "owned local decl but no local module storage");
     return reinterpret_cast<Module *const *>(this)[-1];
   }
-  void setLocalOwningModule(Module *M) {
+  void setLocalOwningModule(const Module *M) {
```

Comment:
This is an example where it's not clear to me that it's correct -- the caller assumes "this will never be mutated" but that's not a safe assumption given `Module *getLocalOwningModule() const;` (which itself is not const correct! that should be a pair of functions where the `const` variant returns a `const Module *` and the non-`const` variant returns a `Module *`)

Reply:
Hmm yes I should have been more careful around that `reinterpret_cast`.